pkg/envoy/embedded_envoy.go

```diff
@@ -63,18 +63,24 @@ func EnableTracing() {
 	tracing = true
 }
 
-func mapLogLevel(level logrus.Level) string {
+func mapLogLevel(agentLogLevel logrus.Level, defaultEnvoyLogLevel string) string {
 	// Set Envoy loglevel to trace if debug AND verbose Engoy logging is enabled
-	if level == logrus.DebugLevel && tracing {
+	if agentLogLevel == logrus.DebugLevel && tracing {
 		return "trace"
 	}
 
 	// Suppress the debug level if not debugging at flow level.
-	if level == logrus.DebugLevel && !flowdebug.Enabled() {
-		level = logrus.InfoLevel
+	if agentLogLevel == logrus.DebugLevel && !flowdebug.Enabled() {
+		return "info"
```

Comment:
nit: Do we want to use the occasion and turn these strings into constants, even in `envoyLevelMap`?

Reply:
good point. changed in the latest commit - thanks!